src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -229,25 +232,96 @@ public static void ValidateEntitiesDoNotGenerateDuplicateQueriesOrMutation(Runti
         }
 
         /// <summary>
-        /// Check whether the entity name defined in runtime config only contains
-        /// characters allowed for GraphQL names.
-        /// Does not perform validation for entities which do not
+        /// Check whether the entity configuration defined in runtime config only contains characters allowed for GraphQL names
+        /// and other validations related to rest path and methods configured for the entity.
+        /// The GraphQL validation is not performed for entities which do not
         /// have GraphQL configuration: when entity.GraphQL == false or null.
         /// </summary>
         /// <seealso cref="https://spec.graphql.org/October2021/#Name"/>
-        /// <param name="entityCollection">The runtime entities to process.</param>
-        public static void ValidateEntityNamesInConfig(RuntimeEntities entityCollection)
+        /// <param name="runtimeConfig">The runtime configuration.</param>
+        public static void ValidateEntityConfiguration(RuntimeConfig runtimeConfig)
         {
-            foreach ((string _, Entity entity) in entityCollection)
+            // Stores the unique rest paths configured for different entities present in the config.
+            HashSet<string> restPathsForEntities = new();
+
+            foreach ((string entityName, Entity entity) in runtimeConfig.Entities)
             {
-                if (entity.GraphQL.Enabled)
+                if (runtimeConfig.Runtime.Rest.Enabled)
+                {
+                    // By default we assume Rest is enabled for the entity.
+                    bool isRestEnabledForEntity = true;
+
+                    // If no custom rest path is defined for the entity, we default it to the entityName.
+                    string pathForEntity = entityName;
+                    if (entity.Rest is not null)
+                    {
+                        pathForEntity = entity.Rest.Path is not null ? entity.Rest.Path.TrimStart('/') : entityName;
+                        isRestEnabledForEntity = entity.Rest.Enabled;
+                        SupportedHttpVerb[] supportedOperations = entity.Rest.Methods;
```

Comment:
is this local variable `supportedOperations` used anywhere in validations? Am I missing something?

Reply:
Yeah correct it isn't.